plugins/tiles/src/main/java/org/apache/struts2/tiles/StrutsApplicationResource.java

```diff
@@ -23,14 +23,41 @@
 import java.io.File;
 import java.io.IOException;
 import java.io.InputStream;
+import java.net.MalformedURLException;
 import java.net.URL;
 
 public class StrutsApplicationResource extends PostfixedApplicationResource {
 
     private final URL url;
 
-    public StrutsApplicationResource(URL url) {
-        super(url.getPath());
+    /**
+     * workarounds WW-5011
+     */
+    private static String getExistedPath(URL url) throws MalformedURLException {
```

Comment:
I named such because WW-5011 is because url.getPath doesn't exist always. What does Extracted refer?

Reply:
I thought "Extracted" was more general - just indicates the method attempts to extract the path from the URL (whether it actually exists or not).
Since the URL resource may not be a file, a File.exists() test cannot always determine if the path actually exists or not...

Reply:
fixed :) - learnt from tiles URLApplicationResource.